backend/models/distance.py

```diff
@@ -151,25 +154,43 @@ def closest_with_early_exit(self, origin, point_list, exit_distance):
         )
 
         if min_haversine_distance <= exit_distance:
+            if self.mode == 'time':
+                return min_haversine_distance / 104607.0 * 60.0, point_list[min_haversine_idx]
             return min_haversine_distance, point_list[min_haversine_idx]
-        else:
-            relevant_points = [
-                point_list[idx]
-                for idx, distance in enumerate(haversine_distances)
-                if distance < min(self.early_exit_outer_radius, 1.5 * min_haversine_distance)
-            ]
-            if not relevant_points:
-                return 10**42, point_list[0]
-            else:
-                return self.closest(origin=origin, point_list=relevant_points)
-
-
-def get_measure(name):
+
+        relevant_points = [
+            point_list[idx]
+            for idx, distance in enumerate(haversine_distances)
+            if distance < min(self.early_exit_outer_radius, 1.5 * min_haversine_distance)
+        ]
+
+        if not relevant_points:
+            return 10**42, point_list[0]
+
+        return self.closest(origin=origin, point_list=relevant_points)
+
+
+class MapBoxDrivingTime(MeasureDistance):
+    """Class for MapBox driving time measurements."""
+
+    def get_distance_in_meters(self, point_a, point_b):
+        """
+        Get haversine distance between two points.
+
+        Expects points as (latitude, longitude) tuples.
+        """
+        if point_a and point_b:
+            return haversine(tuple(point_a), tuple(point_b))
+        return
+
+
+def get_measurer(name):
     """Return an instantiated measure class with the given name."""
-    return MEASURE_NAME_TO_FUNCTION_MAPPING[name.lower()]()
+    return MEASURER_NAME_TO_FUNCTION_MAPPING[name.lower()]()
 
 
-MEASURE_NAME_TO_FUNCTION_MAPPING = {
+MEASURER_NAME_TO_FUNCTION_MAPPING = {
     'haversine': HaversineDistance,
-    'osrm': OSRMDrivingDistance
+    'osrm': OSRMDriving,
```

Comment:
`OSRMDrivingTime` to match `MapBoxDrivingTime`?

Reply:
Same comment as for distance, for now it kinda does both but hopefully split soon.